rsp_vision/dashboard/pages/helpers/calculations_for_plotting.py

```diff
@@ -0,0 +1,162 @@
+import numpy as np
+import pandas as pd
+from scipy.stats import pearsonr
+
+from rsp_vision.analysis.gaussians_calculations import (
+    get_gaussian_matrix_to_be_plotted,
+)
+
+
+def fit_correlation(
+    gaussian: np.ndarray, median_subtracted_response: np.ndarray
+) -> float:
+    """This method calculates the correlation between the median subtracted
+    response and the Gaussian fit (6x6 matrix).
+
+    Parameters
+    ----------
+    gaussian : np.ndarray
+        The Gaussian fit (6x6 matrix).
+    median_subtracted_response : np.ndarray
+        The median subtracted response.
+
+    Returns
+    -------
+    float
+        The correlation between the median subtracted response and the Gaussian
+        fit (6x6 matrix).
+    """
+    fit_corr, _ = pearsonr(
+        median_subtracted_response.flatten(), gaussian.flatten()
+    )
+    return fit_corr
+
+
+def calculate_mean_and_median(
+    signal: pd.DataFrame,
+) -> pd.DataFrame:
+    """This method calculates the mean and median of the signal dataframe
+    over the different stimulus repetitions.
+
+    Parameters
+    ----------
+    signal : pd.DataFrame
+        The signal dataframe containing the data of the ROI. If the direction
+        is pooled, the dataframe contains the data of all directions and the
+        mean and median are calculated over all directions. If the direction
+        is not pooled, the dataframe contains the data of the selected
+        direction and the mean and median are calculated over it.
+        These computations are done in the `sf_tf_grid` callback.
+    Returns
+    -------
+    pd.DataFrame
+        The signal dataframe containing the mean and median of the signal
+        appended to the original dataframe at the end.
+    """
+    mean_df = (
+        signal.groupby(["sf", "tf", "stimulus_frames"])
+        .agg({"signal": "mean"})
+        .reset_index()
+    )
+    mean_df["stimulus_repetition"] = "mean"
+    combined_df = pd.concat([signal, mean_df], ignore_index=True)
+
+    median_df = (
+        signal.groupby(["sf", "tf", "stimulus_frames"])
+        .agg({"signal": "median"})
+        .reset_index()
+    )
+    median_df["stimulus_repetition"] = "median"
+    combined_df = pd.concat([combined_df, median_df], ignore_index=True)
+
+    return combined_df
+
+
+def find_peak_coordinates(
+    fitted_gaussian_matrix: np.ndarray,
+    spatial_frequencies: np.ndarray,
+    temporal_frequencies: np.ndarray,
+    matrix_definition: int,
+) -> tuple:
+    """This method finds the peak coordinates of the fitted gaussian matrix.
```

Comment:
Could just specify here what is meant by coordinates (e.g. tfs, sfs position of the tfs x sfs matrix) or on one of the docstrings, the docstrings are very informative, a quick line to mention what the response matrix is (I read the code before looking at the plots, after the plots it is very clear) e.g. 'Fit a 2-dimensional Gaussian function to a num sfs x num fsf matrix of ROI responses (activity during stimulus presentation minus baseline) to all spatial and temporal frequency combinations'.